core/src/test/java/org/mule/expression/ExpressionManagerTestCase.java

```diff
@@ -141,6 +148,74 @@ public void testBooleanEvaluation()
         assertFalse(muleContext.getExpressionManager().evaluateBoolean("string:false", msg));
     }
 
+    @Test
+    public void testDefaultExpressionFilterActsLikeBooleanValueOfInvalidString()
+    {
+        // Set default value
+        setProperty(MULE_DEFAULT_BOOLEAN_VALUE, "false");
```

Comment:
If the assertion fails, the property will remain set throughout the modules tests. I'd recommend either extracting the tests to different classes and using the `SystemProperty` rule or using a try finally block very carefully.